nas/handler.go

```diff
@@ -36,28 +36,30 @@ func HandleNAS(ue *context.RanUe, procedureCode int64, nasPdu []byte) {
 		} else {
 			if amfSelf.EnableSctpLb {
 				/* checking the guti-ue belongs to this amf instance */
-				id, err := amfSelf.Drsm.FindOwnerInt32ID(ue.AmfUe.Tmsi)
-				if err != nil {
-					logger.NasLog.Errorf("Error checking guti-ue: %v", err)
-				}
-				if id != nil && id.PodName != os.Getenv("HOSTNAME") {
-					rsp := &sdcoreAmfServer.AmfMessage{}
-					rsp.VerboseMsg = "Redirecting Msg From AMF Pod !"
-					rsp.Msgtype = sdcoreAmfServer.MsgType_REDIRECT_MSG
-					rsp.AmfId = os.Getenv("HOSTNAME")
-					/* TODO for this release setting pod ip to simplify logic in sctplb */
-					rsp.RedirectId = id.PodIp
-					rsp.GnbId = ue.Ran.GnbId
-					rsp.Msg = ue.SctplbMsg
-					if ue.AmfUe != nil {
-						ue.AmfUe.Remove()
-					} else {
-						if err := ue.Remove(); err != nil {
-							logger.NasLog.Errorf("Error removing ue: %v", err)
+				if amfSelf.EnableDbStore {
+					id, err := amfSelf.Drsm.FindOwnerInt32ID(ue.AmfUe.Tmsi)
+					if err != nil {
+						logger.NasLog.Errorf("Error checking guti-ue: %v", err)
+					}
+					if id != nil && id.PodName != os.Getenv("HOSTNAME") {
+						rsp := &sdcoreAmfServer.AmfMessage{}
+						rsp.VerboseMsg = "Redirecting Msg From AMF Pod !"
+						rsp.Msgtype = sdcoreAmfServer.MsgType_REDIRECT_MSG
+						rsp.AmfId = os.Getenv("HOSTNAME")
+						/* TODO for this release setting pod ip to simplify logic in sctplb */
+						rsp.RedirectId = id.PodIp
+						rsp.GnbId = ue.Ran.GnbId
+						rsp.Msg = ue.SctplbMsg
+						if ue.AmfUe != nil {
+							ue.AmfUe.Remove()
+						} else {
+							if err := ue.Remove(); err != nil {
+								logger.NasLog.Errorf("Error removing ue: %v", err)
+							}
 						}
+						ue.Ran.Amf2RanMsgChan <- rsp
+						return
 					}
-					ue.Ran.Amf2RanMsgChan <- rsp
-					return
 				}
 			}
```

Comment:
What about the following instead of the nested `if` statements?
```go
				if amfSelf.EnableSctpLb && amfSelf.EnableDbStore {
					...
					...
				}
```